jax/tree_util.py

```diff
@@ -29,6 +26,12 @@
 module-level registry of types, and class hierarchy is ignored. By registering a
 new pytree node type, that type in effect becomes transparent to the utility
 functions in this file.
+
+The primary purpose of this module is to enable the interoperability between
+user defined data structures and JAX-primitives (e.g. `jit`). Its secondary
```

Comment:
Can you change "JAX-primitives" to "JAX transformations" here? Also, I'm not sure I agree with the second sentence; can we delete it, or else discuss so that I can understand what it means?

Reply:
Done. For that sentence I was just trying to rationalize why certain parts of this API is public at all (e.g. `tree_map`) since ostensibly you only need to expose `register_pytree_node` to users. I don't think that's really the confusion I was addressing though, so I removed it.

Reply:
Thanks! It may be that part of my confusion was from use of the word "primitive", because in JAX parlance a primitive is something like `lax.add` or the underlying `lax.add_p`, not the function transformations like `jit`. Moreover the primitives don't know anything about pytrees; all the pytree stuff gets handled in api.py, with the exception of the control flow primitives (because the latter accept Python functions from the user as arguments, making them a bit like transforms).